src/resolver.cpp

```diff
@@ -201,3 +239,41 @@ std::shared_ptr<ArWritableAsset> UsdOpenAssetIOResolver::_OpenAssetForWrite(
 
   return ArDefaultResolver::_OpenAssetForWrite(resolvedPath, writeMode);
 }
+
+std::optional<std::string> UsdOpenAssetIOResolver::locationForEntity(
+    const std::string &assetPath) const {
+  // Check if the assetPath is an OpenAssetIO entity reference.
+  if (auto maybeEntityReference = manager_->createEntityReferenceIfValid(assetPath)) {
+    openassetio::TraitsDataPtr traitsData;
+
+    // Resolve the locateableContent trait in order to get the
+    // (absolute) path to the asset.
+    manager_->resolve(
+        {std::move(*maybeEntityReference)}, {kLocateableContentTraitId}, readContext_,
+        [&traitsData]([[maybe_unused]] std::size_t idx,
+                      const openassetio::TraitsDataPtr &traitsData_) {
```

Comment:
[fix] Can we use a different var name for the lambda - as this is our convention for private members? Makes it hard to grok.

Reply:
Renamed in 1d20438e